### ISSUE TITLE
funannotate compare: protein_ortho 6.0.30 clustering failed with code 33792

### ISSUE THREAD

**Issue by AdamVS** [nextgenusfs/funannotate#604]
Hi Jon,

I am running 1.8.7 with Ubuntu 20.04, 32 cores, ~ 132 GB RAM and python: 3.7.10. It's a mamba installation of funannotate.

**bug description**
The bug is related to proteinortho 6.0.30 and probably earlier versions. Earlier versions of funannotate had a similar problem, and you indicated this might be related to the version of Ubuntu:
[https://github.com/nextgenusfs/funannotate/issues/400](url)

Most recently the code 33792 in proteinortho_clustering for 6.0.30 has crept up again here:
[https://gitlab.com/paulklemm_PHD/proteinortho/-/issues/51](url)

I have not yet rolled back proteinortho from 6.0.30 to 6.0.16 (like here [https://github.com/nextgenusfs/funannotate/issues/450](url)), nor have I yet tried to update proteinortho 6.0.30 in 1.8.7. That is since there is quite a few new installations, updates and downgrades of packages, together with several superseded channel priorities. Perhaps there is an easier solution?

**background**
FYI this is a completely new installation of 1.8.7 so I ran test to check the installation:
`$funannotate test -t all --cpus 8`

Everything tests ok until the late steps of:
`$funannotate compare -i Genome_one.gbk Genome_two.gbk Genome_three.gbk -o compare --cpus 8 --outgroup botrytis_cinerea.dikarya`

**funannotate-compare.log**
```
[Jun 03 07:44 PM]: OS: Ubuntu 20.04, 32 cores, ~ 132 GB RAM. Python: 3.7.10
[Jun 03 07:44 PM]: Running 1.8.7
[Jun 03 07:44 PM]: Now parsing 3 genomes
[Jun 03 07:44 PM]: working on Genome one
[Jun 03 07:44 PM]: working on Genome two
[Jun 03 07:44 PM]: working on Genome three
[Jun 03 07:44 PM]: No secondary metabolite annotations found
[Jun 03 07:44 PM]: Summarizing PFAM domain results
[Jun 03 07:44 PM]: Summarizing InterProScan results
[Jun 03 07:44 PM]: Loading InterPro descriptions
[Jun 03 07:44 PM]: Summarizing MEROPS protease results
[Jun 03 07:44 PM]: found 4 MEROPS familes
/home/user/anaconda3/envs/funannotate/lib/python3.7/site-packages/funannotate/library.py:7866: MatplotlibDeprecationWarning: Calling add_axes() without argument is deprecated since 3.3 and will be removed two minor releases later. You may want to use add_subplot() instead.
cbar_ax = fig.add_axes(shrink=0.4)
[Jun 03 07:44 PM]: Summarizing CAZyme results
[Jun 03 07:44 PM]: found 5 CAZy familes
/home/user/anaconda3/envs/funannotate/lib/python3.7/site-packages/funannotate/library.py:7866: MatplotlibDeprecationWarning: Calling add_axes() without argument is deprecated since 3.3 and will be removed two minor releases later. You may want to use add_subplot() instead.
cbar_ax = fig.add_axes(shrink=0.4)
[Jun 03 07:44 PM]: Summarizing COG results
[Jun 03 07:44 PM]: Summarizing secreted protein results
[Jun 03 07:44 PM]: Summarizing fungal transcription factors
/home/user/anaconda3/envs/funannotate/lib/python3.7/site-packages/funannotate/library.py:7866: MatplotlibDeprecationWarning: Calling add_axes() without argument is deprecated since 3.3 and will be removed two minor releases later. You may want to use add_subplot() instead.
cbar_ax = fig.add_axes(shrink=0.4)
[Jun 03 07:44 PM]: Running GO enrichment for each genome
WARNING: skipping Genome_one.txt as no GO terms
[Jun 03 07:51 PM]: Running orthologous clustering tool, ProteinOrtho. This may take awhile...
[Jun 03 07:51 PM]: CMD ERROR: proteinortho -project=funannotate -synteny -cpus=8 -singles -selfblast Genome_one.faa Genome_two.faa Genome_three.faa
[Jun 03 07:51 PM]: *****************************************************************
Proteinortho with PoFF version 6.0.30 - An orthology detection tool

Using 8 CPU threads, Detected 'diamond' version 2.0.8
Checking input files.
Checking Genome_one.faa... Genome_one.faa 124 genes ok
Checking Genome_two.faa... Genome_two.faa 295 genes ok
Checking Genome_three.faa... Genome_three.faa 172 genes ok

Step 1
Generating indices.
Building database for 'Genome_two.faa' (295 sequences)
Building database for 'Genome_three.faa' (172 sequences)
Building database for 'Genome_one.faa' (124 sequences)

Step 2 using diamond with : synteny selfblast
Running blast analysis: 100% (6/6)
[OUTPUT] -> written to funannotate.blast-graph

Step 2.5 Checking blast graph(s)
funannotate.ffadj-graph is free of duplicated edges

Step 3
Clustering by similarity (Proteinortho mode) and 8 cpu core(s).
[WARNING] proteinortho_clustering failed. I will now retry without the OMP_PROC_BIND flag.

Parameter-vector : (version=6.0.30,step=0,verbose=1,debug=1,exactstep3=0,synteny=1,duplication=2,cs=3,alpha=0.5,connectivity=0.1,cpus=8,evalue=1e-05,purity=1e-07,coverage=50,identity=25,blastmode=diamond,sim=0.95,report=3,keep=0,force=0,selfblast=1,twilight=0,singles=1,clean=0,blastOptions=,nograph=0,xml=0,desc=0,tmp_path=./proteinortho_cache_funannotate/,blastversion=2.0.8,binpath=,makedb=diamond makedb --in,blast=,jobs_todo=6,project=funannotate,po_path=/home/adamvs/anaconda3/envs/funannotate/bin/,run_id=,threads_per_process=1,useMcl=0,freemem=96639)

[Error] 'proteinortho_clustering' failed with code 33792. (Please visit https://gitlab.com/paulklemm_PHD/proteinortho/wikis/Error%20Code)
Maybe your operating system does not support the statically compiled version, please try recompiling proteinortho with 'make clean' and 'make' (and 'make install PREFIX=...').

If you cannot solve this error, please send a report to incoming+paulklemm-phd-proteinortho-7278443-issue-@incoming.gitlab.com including the parameter-vector above or visit https://gitlab.com/paulklemm_PHD/proteinortho/wikis/Error%20Codes for more help.
Further more all mails to lechner@staff.uni-marburg.de are welcome

#########################################################
ERROR: funannotate compare test failed - check logfiles
#########################################################
```

**versions and dependencies**
```
~/funannotate_db$ funannotate check --show-versions
You are running Python v 3.7.10. Now checking python packages...
biopython: 1.78
goatools: 1.1.5
matplotlib: 3.4.2
natsort: 7.1.1
numpy: 1.20.3
pandas: 1.2.4
psutil: 5.8.0
requests: 2.25.1
scikit-learn: 0.24.2
scipy: 1.6.3
seaborn: 0.11.1
All 11 python packages installed

You are running Perl v b'5.026002'. Now checking perl modules...
Bio::Perl: 1.007002
Carp: 1.38
Clone: 0.42
DBD::SQLite: 1.64
DBD::mysql: 4.046
DBI: 1.642
DB_File: 1.855
Data::Dumper: 2.173
File::Basename: 2.85
File::Which: 1.23
Getopt::Long: 2.5
Hash::Merge: 0.300
JSON: 4.02
LWP::UserAgent: 6.39
Logger::Simple: 2.0
POSIX: 1.76
Parallel::ForkManager: 2.02
Pod::Usage: 1.69
Scalar::Util::Numeric: 0.40
Storable: 3.15
Text::Soundex: 3.05
Thread::Queue: 3.12
Tie::File: 1.02
URI::Escape: 3.31
YAML: 1.29
threads: 2.15
threads::shared: 1.56
All 27 Perl modules installed

Checking Environmental Variables...
$FUNANNOTATE_DB=/home/adamvs/funannotate_db
$PASAHOME=/home/adamvs/anaconda3/envs/funannotate/opt/pasa-2.4.1
$TRINITY_HOME=/home/adamvs/anaconda3/envs/funannotate/opt/trinity-2.8.5
$EVM_HOME=/home/adamvs/anaconda3/envs/funannotate/opt/evidencemodeler-1.1.1
$AUGUSTUS_CONFIG_PATH=/home/adamvs/anaconda3/envs/funannotate/config/
$GENEMARK_PATH=/home/adamvs/bioinf/GeneMark/gmes_linux_64
All 6 environmental variables are set

Checking external dependencies...
PASA: 2.4.1
CodingQuarry: 2.0
Trinity: 2.8.5
augustus: 3.3.3
bamtools: bamtools 2.5.1
bedtools: bedtools v2.30.0
blat: BLAT v36
diamond: 2.0.8
emapper.py: 2.1.2
ete3: 3.1.2
exonerate: exonerate 2.4.0
fasta: no way to determine
glimmerhmm: 3.0.4
gmap: 2017-11-15
gmes_petap.pl: 4.65_lic
hisat2: 2.2.1
hmmscan: HMMER 3.3.2 (Nov 2020)
hmmsearch: HMMER 3.3.2 (Nov 2020)
java: 11.0.8-internal
kallisto: 0.46.1
mafft: v7.480 (2021/May/21)
makeblastdb: makeblastdb 2.2.31+
minimap2: 2.20-r1061
proteinortho: 6.0.30
pslCDnaFilter: no way to determine
salmon: salmon 0.14.1
samtools: samtools 1.10
signalp: 5.0b
snap: 2006-07-28
stringtie: 2.1.5
tRNAscan-SE: 2.0.7 (Oct 2020)
tantan: tantan 26
tbl2asn: no way to determine, likely 25.X
tblastn: tblastn 2.2.31+
trimal: trimAl v1.4.rev15 build[2013-12-17]
trimmomatic: 0.39
All 36 external dependencies are installed
```

**Comment by nextgenusfs** [comment]
Def looks like the compilation issue I've seen before where clustering does not work -- the author isn't able to repeat apparently on a docker setup, but I've seen it on multiple systems.  I would just pin it to 6.0.16 and see if that fixes it.  Would be nice if this was fixed upstream, but I tried ~6 months ago and didn't get anywhere.

**Comment by AdamVS** [comment]
Hi Jon,

The proteinortho issue is now resolved for me with `proteinortho: 6.0.31` and my test with `funannotate test -t all --cpus 8` works as expected. Issue as resolved here [https://gitlab.com/paulklemm_PHD/proteinortho/-/issues/51](url)

many thanks,
Adam
